lecture_1/homework/app/controllers/api/v1/samurais_controller.rb

```diff
@@ -0,0 +1,69 @@
+module Api
+  module V1
+    class SamuraisController < ApplicationController
+      before_action :default_limit, only: %i[index]
+
+      def show
+        render json: samurai.to_json(only: samurai_attributes)
+      end
+
+      def index
+        samurais = samurais_status
+        render json: samurais.to_json(only: samurai_attributes)
+      end
+
+      def create
+        samurai = clan.samurais.new(samurai_params)
+
+        if samurai.save
+          render json: samurai.to_json(only: samurai_attributes), status: 201
+        else
+          render json: { errors: samurai.errors.messages }, status: 422
+        end
+      end
+
+      def update
+        if samurai.update(samurai_params)
+          render json: samurai.to_json(only: samurai_attributes)
+        else
+          render json: { errors: samurai.errors.messages }, status: 422
+        end
+      end
+
+      def destroy
+        samurai.destroy!
+
+        head 204
+      end
+
+      private
+
+      def samurai
+        @samurai ||= clan.samurais.find(params[:id])
+      end
+
+      def clan
+        @clan ||= Clan.find(params[:clan_id])
+      end
+
+      def samurai_params
+        params.permit(samurai_attributes)
+      end
+
+      def samurai_attributes
+        %w[id name armor battle_count join_at death_at clan_id]
+      end
+
+      def samurais_status
+        case ActiveModel::Type::Boolean.new.cast(params[:alive])
```

Comment:
Nie wydaje mi się, żeby takie castowanie było potrzebne, ogółem bezpieczniej by było w tym case skorzystać z `to_i` i próbować załatwić to intami, `case when` bywają trochę shady, bo korzystają z operatorów `===` i `or`, a nie każdy sobie zdaje z tego sprawę :D Nie mniej, Twoje rozwiązanie nadal jest spoko :)

Reply:
Napisałem trochę inaczej :)